### DISCUSSION TITLE
Reduce bundle size by eliminating state management

### DISCUSSION THREAD

**Discussion by alshdavid** [vuejs/petite-vue#127] [General]
This project is incredible, love it. I have been using this in conjunction with W3C custom elements and I have found it to be a very ergonomic and bloat-free way to write web applications.

Bundle size is a huge factor for me so I was wondering how much of an impact the built-in state management has on bundle size/parse time (it's currently 17kb uncompressed).

I tend to compose my applications by brining together features. petite-vue is exclusively as a renderer for my presentation layer, where having state management within my view renderer is not very useful to me.

**Comment by nfplee** [comment]
Any chance you could share how to create a petite-vue component as a web component (custom element)?

**Reply by niutech** [reply]
[Here you go](https://github.com/vuejs/petite-vue/discussions/130)

**Comment by niutech** [comment]
I don't think 17KB uncompressed (7 KB gzipped) is an issue, especially when it is cached by the browser, so it is downloaded only once. That said, you don't need Petite Vue at all, you could make your own presentation layer using vanilla JS.
